language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/CommandsSubmission.java

```diff
@@ -0,0 +1,246 @@
+ *                                   .with...
+ * <pre/>
+ */
+public class CommandsSubmission {
+  private String applicationId;
+  private String commandId;
+  private List<@NonNull ? extends HasCommands> commands;
+
+  private String workflowId;
+  private List<@NonNull String> actAs;
+  private List<@NonNull String> readAs;
+  private Optional<Instant> minLedgerTimeAbs;
+  private Optional<Duration> minLedgerTimeRel;
+  private Optional<Duration> deduplicationTime;
+  private Optional<String> accessToken;
+
+  private CommandsSubmission(
+      String workflowId,
+      String applicationId,
+      String commandId,
+      List<@NonNull String> actAs,
+      List<@NonNull String> readAs,
+      Optional<Instant> minLedgerTimeAbs,
+      Optional<Duration> minLedgerTimeRel,
+      Optional<Duration> deduplicationTime,
+      List<@NonNull ? extends HasCommands> commands,
+      Optional<String> accessToken) {
+    this.workflowId = workflowId;
+    this.applicationId = applicationId;
+    this.commandId = commandId;
+    this.actAs = actAs;
+    this.readAs = readAs;
+    this.minLedgerTimeAbs = minLedgerTimeAbs;
+    this.minLedgerTimeRel = minLedgerTimeRel;
+    this.deduplicationTime = deduplicationTime;
+    this.commands = commands;
+    this.accessToken = accessToken;
+  }
+
+  public static CommandsSubmission create(
+      String applicationId, String commandId, List<@NonNull ? extends HasCommands> commands) {
+    return new CommandsSubmission(
+        "",
+        applicationId,
+        commandId,
+        emptyList(),
+        emptyList(),
+        empty(),
+        empty(),
+        empty(),
+        commands,
+        empty());
+  }
+
+  public String getWorkflowId() {
+    return workflowId;
+  }
+
+  public String getApplicationId() {
+    return applicationId;
+  }
+
+  public String getCommandId() {
+    return commandId;
+  }
+
+  public List<String> getActAs() {
+    return actAs;
+  }
+
+  public List<String> getReadAs() {
+    return readAs;
+  }
+
+  public Optional<Instant> getMinLedgerTimeAbs() {
+    return minLedgerTimeAbs;
+  }
+
+  public Optional<Duration> getMinLedgerTimeRel() {
+    return minLedgerTimeRel;
+  }
+
+  public Optional<Duration> getDeduplicationTime() {
+    return deduplicationTime;
+  }
+
+  public List<? extends HasCommands> getCommands() {
+    return commands;
+  }
+
+  public Optional<String> getAccessToken() {
+    return accessToken;
+  }
+
+  public CommandsSubmission withWorkflowId(String workflowId) {
+    return new CommandsSubmission(
+        workflowId,
+        applicationId,
+        commandId,
+        actAs,
+        readAs,
+        minLedgerTimeAbs,
+        minLedgerTimeRel,
+        deduplicationTime,
+        commands,
+        accessToken);
+  }
+
+  public CommandsSubmission withActAs(String actAs) {
+    return new CommandsSubmission(
+        workflowId,
+        applicationId,
+        commandId,
+        singletonList(actAs),
```

Comment:
I believe `singletonList` is frozen. Is this something that you want? Otherwise if you want to use more modern Java feature you can call `List.of(actAs)` instead.